File: src/main/java/insogroup/HtmlParser/StandardRealisation/PluginSyntaxAnalyzer/interfaces/State.java
package insogroup.HtmlParser.StandardRealisation.PluginSyntaxAnalyzer.interfaces;

import insogroup.HtmlParser.StandardRealisation.exceptions.PluginException;
import org.xml.sax.Attributes;

import java.util.HashMap;
import java.util.List;

public interface State {

    State addChild(String name, Attributes attr);

    String get(String key) throws PluginException;

    State getParent() throws PluginException;

    String getTag();

    String getText();

    void setText(String text);

    List<State> getChilds();

    State getRoot();

    HashMap<String, String> getAttributes();
    
}

File: src/main/kotlin/com/github/insanusmokrassar/HTMLParser/StandardRealisation/PluginSyntaxAnalyzer/interfaces/State.kt
package com.github.insanusmokrassar.HTMLParser.StandardRealisation.PluginSyntaxAnalyzer.interfaces

import org.xml.sax.Attributes

interface State {

    val parent: State?

    val tag: String

    var text: String

    val childs: List<State>

    val root: State

    val attributes: Map<String, String>

    fun addChild(name: String, attr: Attributes): State

    operator fun get(key: String): String
}
